app/src/main/java/net/squanchy/proximity/near/NearITProximityProvider.java

```diff
@@ -4,31 +4,47 @@
 import android.util.Log;
 
 import net.squanchy.R;
+import net.squanchy.proximity.ProximityEvent;
 import net.squanchy.proximity.ProximityProvider;
 
+import io.reactivex.Observable;
+import io.reactivex.ObservableEmitter;
+import io.reactivex.schedulers.Schedulers;
 import it.near.sdk.Geopolis.Beacons.Ranging.ProximityListener;
 import it.near.sdk.NearItManager;
+import it.near.sdk.Reactions.CustomJSON.CustomJSON;
 
-public class NearITProximityProvider implements ProximityProvider{
+public class NearITProximityProvider implements ProximityProvider {
 
-    NearItManager nearItManager;
+    private final NearItManager nearItManager;
+    private static final String PROXIMITY_ACTION = "action";
+    private static final String SUBJECT_ID = "subject";
 
-    public NearITProximityProvider(Context context) {
-        Log.d("NearITProximityProvider", "new instance created");
-        nearItManager = new NearItManager(context.getApplicationContext(), context.getString(R.string.nearit_api_key));
+    public NearITProximityProvider(NearItManager nearItManager) {
+        this.nearItManager = nearItManager;
         // TODO set a notification icon if needed
     }
 
-    public void registerForProximityEvents(ProximityListener listener) {
-        nearItManager.addProximityListener(listener);
-    }
-
-    public void deregisterForProximityEvents(ProximityListener listener) {
-        nearItManager.removeProximityListener(listener);
-    }
-
     @Override
     public void startRadar() {
         nearItManager.startRadar();
```

Comment:
Question: is there a way to "stop" the radar as well? I can see how this could go inside the observable creation (so that you don't need to call 2 methods in sequence) and you could stop the radar when the observable unsubscribes. But maybe I didn't understand how the NearIT SDK works and what I said doesn't make sense :speak_no_evil: (e.g. in the case `startRadar` has to be some sort of global start).